### DISCUSSION TITLE
save a shared variable in node

### DISCUSSION THREAD

**Discussion by MrZmcH** [eclipse-theia/theia#13186] [Q&A]
In theia, how do you save a shared variable in node, similar to LocalStorageService in your browser?

**Comment by msujew** [comment]
Theia usually stores semi-permanent information in the `~/.theia` directory. You should probably do the same.

**Reply by MrZmcH** [reply]
@msujew I don't know what it means. How do I use it? Do you have examples?

**Reply by msujew** [reply]
It's literally just a folder in your system. `~/.theia` is just a folder named `.theia` in your user directory. You can take a look at that in your own system. A short code snippet would look like this:

```ts
import * as path from 'path';
import * as os from 'os';

const filePath = path.join(os.homedir(), '.theia', '<filename>');
```

**Comment by JonasHelming** [comment]
@MrZmcH Does this answer your question? If so, would you mind marking it as answer, pleae?
